version/ua.go

```diff
@@ -0,0 +1,63 @@
+package version
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"sync"
+)
+
+var (
+	reRelease *regexp.Regexp
+	reDev     *regexp.Regexp
+	reOnce    sync.Once
+	uapCbs    map[string]func() string
+	uapCached map[string]string
+	uapMu     sync.RWMutex
```

Comment:
I'm fine with having skipping the `uapMu` and adding comment to `Set` instead. It would be called from `init`.

`uapCached` can be skipped as well. Called side can manage the caching if it needs to.